app/Http/Controllers/PromotionController.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use App\Promotion;
+use Validator;
+
+class PromotionController extends Controller
+{
+    /**
+     * Display all promotions of the all bussiness.
+     *
+     * @return Response
+     */
+    public function index()
+    {
+        $data = Promotion::all();
+        return view('promotion.list', ['data' => $data]);
+    }
+
+    /**
+     * Display form add promotion.
+     *
+     * @return Response
+     */
+    public function showAddForm()
+    {
+        return view('promotion.create');
+    }
+
+    /**
+     * Store a promotion is created into promotions table.
+     *
+     * @param Request $request Validate value input
+     *
+     * @return Response
+     */
+    public function store(Request $request)
+    {
+        $validate = Validator::make($request->all(), [
+                    'title' => 'required|unique:promotions,title',
+                    'description' => 'required',
+                    'percent' => 'required',
+                    'quantity' => 'required',
+                    'start-date' => 'required',
+                    'end-date' => 'required',
+        ]);
+        if ($validate->fails()) {
+            return redirect()->back()->withErrors($validate->errors())->withInput();
+        } else {
+            $data = new Promotion;
+            $data->title = $request->input('title');
+            $data->description = $request->input('description');
+            $data->percent = $request->input('percent');
+            $data->quantity = $request->input('quantity');
+            $data->date_start = $request->input('start-date');
+            $data->date_end = $request->input('end-date');
+            $data->product_id = $request->input('product-id');
+            $data->save();
+            return redirect()->route('list');
+        }
+    }
+
+    /**
+     * Display form edit a promotion selected.
+     *
+     * @param Int $id Get value input
+     *
+     * @return Response
+     */
+    public function showEditForm($id)
```

Comment:
[rank C] I think this function name is wrong. We need follow function name as resource route